cmd/init.go

```diff
@@ -307,8 +307,15 @@ func install(config *initCommandConfig) error {
 	config.Info.log("Setting up the development environment")
 
 	// reset config.StackRegistry and get it again from the newly untarred .appsody-config.yaml
-	// this will need to be updated when we support --stack-registry on the init command
-	config.StackRegistry = ""
+
+	if config.StackRegistry != "" {
```

Comment:
I'm not convinced this is going to work. What if the stack template's `.appsody-config.yaml` already fully qualifies the stack registry and the user doesn't set `--stack-registry` flag? In this case we shouldn't modify the file, but I'm worried this code will change it.

Reply:
I've done some significant refactoring - please take a look.